module/mempool/stdmap/incorporated_result_seals_test.go

```diff
@@ -3,12 +3,191 @@ package stdmap
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"pgregory.net/rapid"
+
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
+// icrSealsMachine is a description of a state machine for testing IncorporatedresultSeals
+type icrSealsMachine struct {
+	icrs  *IncorporatedResultSeals       // icrSeals being tested
+	state []*flow.IncorporatedResultSeal // model of the icrSeals
+	limit uint
+}
+
+// Init is an action for initializing a icrSeals instance.
+func (m *icrSealsMachine) Init(t *rapid.T) {
+	n := uint(rapid.IntRange(1, 1000).Draw(t, "n").(int))
+	m.icrs = NewIncorporatedResultSeals(n)
+	m.limit = n
+}
+
+// Add is a conditional action which adds an item to the icrSeals.
+func (m *icrSealsMachine) Add(t *rapid.T) {
+	i := rapid.Uint64().Draw(t, "i").(uint64)
+
+	seal := unittest.IncorporatedResultSeal.Fixture(func(s *flow.IncorporatedResultSeal) {
+		s.Header.Height = i
+	})
+
+	_, err := m.icrs.Add(seal)
+	require.NoError(t, err)
+
+	unmet := true
+	for _, v := range m.state {
+		if v.ID() == seal.ID() {
+			unmet = false
+		}
+	}
+
+	if unmet && seal.Header.Height >= m.icrs.lowestHeight {
+		m.state = append(m.state, seal)
+		if len(m.state) > int(m.limit) {
+			// we remove one of the max height elements if we're beyond the mempool limit
+			max_height := m.icrs.lowestHeight
+			for _, v := range m.state {
+				if v.Header.Height > max_height {
+					max_height = v.Header.Height
+				}
+			}
+			filtered_state := make([]*flow.IncorporatedResultSeal, 0)
+			for _, v := range m.state {
+				if v.Header.Height < max_height {
+					filtered_state = append(filtered_state, v)
+				} else {
+					// the actual removed elements are chosen at random by the backend,
+					// so we probe for all that verify the height constraint
+					_, still_there := m.icrs.ByID(v.ID())
+					if still_there {
+						filtered_state = append(filtered_state, v)
+					}
+				}
+			}
+			m.state = filtered_state
+		}
+	}
+}
+
+// Prune is a Conditional action that removes elements of height strictly lower than its argument
+func (m *icrSealsMachine) PruneUpToHeight(t *rapid.T) {
+	h := rapid.Uint64().Draw(t, "h").(uint64)
+	err := m.icrs.PruneUpToHeight(h)
+	if h >= m.icrs.lowestHeight {
+		require.NoError(t, err)
+		assert.Equal(t, m.icrs.lowestHeight, h)
+	}
+
+	filtered_state := make([]*flow.IncorporatedResultSeal, 0)
+	for _, v := range m.state {
+		if v.Header.Height >= h {
+			filtered_state = append(filtered_state, v)
+		}
+	}
+	m.state = filtered_state
+}
+
+// Get is an action that retrieves an element from the icrSeals
+func (m *icrSealsMachine) Get(t *rapid.T) {
+	n := len(m.state)
+	// skip if the store is empty
+	if n > 0 {
+		i := rapid.IntRange(0, n-1).Draw(t, "i").(int)
+
+		s := m.state[i]
+		actual, ok := m.icrs.ByID(s.ID())
+		require.True(t, ok)
+		require.Equal(t, s, actual)
+	}
```

Comment:
```suggestion
	n := len(m.state)
	// skip if the store is empty
	if n == 0 {
		return
	}
	
	i := rapid.IntRange(0, n-1).Draw(t, "i").(int)
	s := m.state[i]
	actual, ok := m.icrs.ByID(s.ID())
	require.True(t, ok)
	require.Equal(t, s, actual)
	}
```

Another convention/style thing: if possible we try to indent exit conditions or edge cases and keep the main logic flow un-indented.